cmd/otelcorecol/go.mod

```diff
@@ -2,7 +2,7 @@
 
 module go.opentelemetry.io/collector/cmd/otelcorecol
 
-go 1.21
+go 1.21.0
```

Comment:
Not sure about this change, hope it will not trip the CI.

Reply:
weird, will revert

Reply:
```suggestion
go 1.21
```